tests/test_flowdiagnosticstool.cpp

```diff
@@ -439,4 +439,198 @@ BOOST_AUTO_TEST_CASE (OneDimCase)
+                          ConnectionValues::NumPhases     { 1 });
+    const size_t nconn = cas.connectivity().numConnections();
+    for (size_t conn = 0; conn < nconn; ++conn) {
+        BOOST_TEST_MESSAGE("Connection " << conn << " connects cells "
+                           << graph.connection(conn).first << " and "
+                           << graph.connection(conn).second);
+    }
+
+    using C = ConnectionValues::ConnID;
+    using P = ConnectionValues::PhaseID;
+    flux(C{0}, P{0}) = 0.3;
+    flux(C{1}, P{0}) = 0.0;
+    flux(C{2}, P{0}) = 0.3;
+    flux(C{3}, P{0}) = 0.6;
+    flux(C{4}, P{0}) = 0.0;
+    flux(C{5}, P{0}) = 0.3;
+    flux(C{6}, P{0}) = -0.6;
+
+    // Create well in/out flows.
+    CellSetValues wellflow = { {0, 0.3}, {3, 0.3}, {2, -0.6} };
+
+    Toolbox diagTool(graph);
+    diagTool.assignPoreVolume(cas.poreVolume());
+    diagTool.assignConnectionFlux(flux);
+    diagTool.assignInflowFlux(wellflow);
+
+    auto injstart = std::vector<CellSet>{ CellSet(CellSetID("I-1"), {0}),
+                                          CellSet(CellSetID("I-2"), {3}) };
+    auto prdstart = std::vector<CellSet>{ CellSet(CellSetID("P-1"), {2}) };
+
+    const auto fwd = diagTool.computeInjectionDiagnostics(injstart);
+    const auto rev = diagTool.computeProductionDiagnostics(prdstart);
+
+    // Global ToF field (accumulated from all injectors)
+    {
+        const auto tof = fwd.fd.timeOfFlight();
+
+        BOOST_REQUIRE_EQUAL(tof.size(), cas.connectivity().numCells());
+        std::vector<double> expected = { 1.0, 2.0, 3.0, 1.0, 2.0, 2.5 };
+        check_is_close(tof, expected);
+    }
+
+    // Global ToF field (accumulated from all producers)
+    {
+        const auto tof = rev.fd.timeOfFlight();
+
+        BOOST_REQUIRE_EQUAL(tof.size(), cas.connectivity().numCells());
+        std::vector<double> expected = { 3.5, 2.5, 0.5, 2.5, 1.5, 1.0 };
+        check_is_close(tof, expected);
+    }
+
+    // Verify set of start points.
+    {
+        using VCS = std::vector<Opm::FlowDiagnostics::CellSet>;
+        using VCSI = std::vector<Opm::FlowDiagnostics::CellSetID>;
+        using P = std::pair<VCS, VCSI>;
+        std::vector<P> pairs { P{ injstart, fwd.fd.startPoints() }, P{ prdstart, rev.fd.startPoints() } };
+        for (const auto& p : pairs) {
+            const auto& s1 = p.first;
+            const auto& s2 = p.second;
+            BOOST_CHECK_EQUAL(s1.size(), s2.size());
+            for (const auto& pt : s2) {
+                // ID of 'pt' *MUST* be in set of identified start points.
+                auto pos = std::find_if(s1.begin(), s1.end(),
+                                        [&pt](const CellSet& s)
+                                        {
+                                            return s.id().to_string() == pt.to_string();
+                                        });
+                BOOST_CHECK(pos != injstart.end());
```

Comment:
Did you mean
```C++
BOOST_CHECK(pos != s1.end());
```
here?  Otherwise, I don't understand how the check could be meaningful for `pairs[1]`.